src/renderer/components/Editor/TakeComponent.tsx

```diff
@@ -111,68 +121,71 @@ const TakeComponent = ({
     setIsFirstTimeOpen,
   ]);
 
+  const NumberedButton = (
+    <Avatar
+      onClick={onSelectTake}
+      sx={{
+        height: 20,
+        width: 20,
+        fontSize: 12,
+        color: colors.grey[700],
+        backgroundColor: isActive ? colors.yellow[500] : colors.grey[400],
+        display: 'flex',
+        position: 'absolute',
+        left: '-30px',
+        transform: 'translateY(2px)',
+        cursor: 'pointer',
+      }}
+    >
+      {takeIndex + 1}
+    </Avatar>
+  );
+
+  const TakeWords = takeWords.map((word, index, words) => {
+    const wordIndex = transcriptionIndex + index;
+    return (
+      <WordOuterComponent
+        key={`word-outer-${word.originalIndex}-${word.pasteKey}`}
+        word={word}
+        prevWord={wordIndex > 0 ? words[wordIndex - 1] : null}
+        nextWord={wordIndex < words.length - 1 ? words[wordIndex + 1] : null}
+        index={wordIndex}
+        isPlaying={nowPlayingWordIndex === wordIndex}
+        isPrevWordSelected={isIndexInRange(selection, wordIndex - 1)}
+        isSelected={isIndexInRange(selection, wordIndex)}
+        isNextWordSelected={isIndexInRange(selection, wordIndex + 1)}
+        onMouseDown={onWordMouseDown}
+        onMouseEnter={onWordMouseEnter}
+        isInInactiveTake={!(isActive || isTakeGroupOpened) && !isFirstTimeOpen}
+        transcriptionLength={words.length}
+        {...passThroughProps}
+      />
+    );
+  });
+
+  useLayoutEffect(() => {
```

Comment:
What happens if this is a regular `useEffect`? Is there a flicker?

Reply:
I think it looks a little more smoother with useLayoutEffect but they're both not perfect